src/hb-ot-color.h

```diff
@@ -0,0 +1,156 @@
+ */
+
+#ifndef HB_OT_H_IN
+#error "Include <hb-ot.h> instead."
+#endif
+
+#ifndef HB_OT_COLOR_H
+#define HB_OT_COLOR_H
+
+#include "hb.h"
+
+#include "hb-ot-tag.h"
+
+HB_BEGIN_DECLS
+
+/**
+ * HB_OT_TAG_CPAL:
+ * a four-letter tag for identifying the CPAL table with color palettes
+ *
+ * Since: 1.2.8
+ */
+#define HB_OT_TAG_CPAL HB_TAG('C','P','A','L')
+
+
+/**
+ * hb_ot_color_t:
+ * @red: the intensity of the red channel
+ * @green: the intensity of the green channel
+ * @blue: the intensity of the blue channel
+ * @alpha: the transparency
+ *
+ * Structure for holding color values.
+ *
+ * Since: 1.2.8
+ */
+typedef struct
+{
+  uint8_t red, green, blue, alpha;
+} hb_ot_color_t;
+
+
+/**
+ * hb_ot_color_palette_flags_t:
+ * @HB_OT_COLOR_PALETTE_FLAG_DEFAULT: default indicating that there is nothing special to note about a color palette.
+ * @HB_OT_COLOR_PALETTE_FLAG_FOR_LIGHT_BACKGROUND: flag indicating that the color palette is suitable for rendering text on light background.
+ * @HB_OT_COLOR_PALETTE_FLAG_FOR_DARK_BACKGROUND: flag indicating that the color palette is suitable for rendering text on dark background.
+ *
+ * Since: 1.2.8
+ */
+typedef enum { /*< flags >*/
+  HB_OT_COLOR_PALETTE_FLAG_DEFAULT = 0x00000000u,
+  HB_OT_COLOR_PALETTE_FLAG_FOR_LIGHT_BACKGROUND = 0x00000001u,
+  HB_OT_COLOR_PALETTE_FLAG_FOR_DARK_BACKGROUND = 0x00000002u,
+} hb_ot_color_palette_flags_t;
+
+
+/**
+ * hb_ot_color_get_palette_count:
+ * @face: a font face
+ *
+ * Returns: the number of color palettes in @face, or zero if @face has
+ * no color
+ *
+ * Since: 1.2.8
+ */
+HB_EXTERN unsigned int
+hb_ot_color_get_palette_count (hb_face_t *face);
+
+
+/**
+ * hb_ot_color_get_palette_name:
+ * @face: a font face
+ * @palette: the index of the color palette whose name is being requested
+ *
+ * Retrieves the name of a color palette. For example, a color font can
+ * have themed palettes like "Spring", "Summer", "Fall", and "Winter".
+ *
+ * Returns: an identifier within @face's `name` table.
+ * If the requested palette has no name, or if @face has no color,
+ * or if @palette is not between 0 and hb_ot_color_get_palette_count(),
+ * the result is zero.
+ *
+ * Since: 1.2.8
+ */
+HB_EXTERN unsigned int
+hb_ot_color_get_palette_name (hb_face_t *face, unsigned int palette);
+
+
+/**
+ * hb_ot_color_get_palette_flags:
+ * @face: a font face
+ * @palette: the index of the color palette whose flags are being requested
+ *
+ * Returns: the flags for the requested color palette.  If @face has no color,
+ * or if @palette is not between 0 and hb_ot_color_get_palette_count(),
+ * the result is #HB_OT_COLOR_PALETTE_FLAG_DEFAULT.
+ *
+ * Since: 1.2.8
+ */
+HB_EXTERN hb_ot_color_palette_flags_t
+hb_ot_color_get_palette_flags (hb_face_t *face, unsigned int palette);
+
+
+/**
+ * hb_ot_color_get_palette_colors:
+ * @face: a font face
+ * @palette: the index of the color palette whose colors are being requested
+ * @start_offset: the index of the first color being requested.
```

Comment:
This should be annotated with `(in out)` for introspection (see https://wiki.gnome.org/Projects/GObjectIntrospection/Annotations).